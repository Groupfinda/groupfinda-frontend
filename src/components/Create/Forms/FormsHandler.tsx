import React, { useState } from "react";
import { Layout } from "@ui-kitten/components";
import { StyleSheet } from "react-native";
import EventDetailsForm from "./EventDetailsForm";
import EventDateForm from "./EventDateForm";
import EventLocationForm from "./EventLocationForm";
import EventImagesForm from "./EventImagesForm";
import EventCategoryForm from "./EventCategoryForm";
import SubmitForm from "./SubmitForm";
import { FormVariablesType } from "./types";
import { CustomError } from "../../../hooks";
import { useApolloClient } from "@apollo/react-hooks";
import { ME, GetEditEventData } from "../../../graphql/queries";

type Props = {
  event: GetEditEventData["getEvent"] | undefined;
};

const FormsHandler: React.FC<Props> = (props) => {
  let initialVariables: FormVariablesType = {
    title: "",
    description: "",
    dateOfEvent: new Date(),
    recurringMode: false,
    dateLastRegister: new Date(),
    images: [],
    privateStatus: true,
    groupSize: 4,
    category: [],
    locationOn: false,
    address: "",
    postalCode: "",
  };
  const event = props.event;

  if (event) {
    const {
      title,
      description,
      dateOfEvent,
      dateLastRegister,
      images,
      groupSize,
      category,
      location,
    } = event;
    initialVariables = {
      ...initialVariables,
      title,
      description,
      dateOfEvent: new Date(dateOfEvent),
      dateLastRegister: new Date(dateLastRegister),
      images,
      groupSize,
      category,
      address: location.address,
      postalCode: location.postalCode,
      privateStatus: event.private,
    };
  }

  const [page, setPage] = useState<number>(0);
  const [variables, setVariables] = useState<FormVariablesType>(
    initialVariables
  );

  const client = useApolloClient();
  const userQuery = client.readQuery({ query: ME });
  const role = userQuery.me.role;

  const nextPage = () => {
    setPage(page + 1);
  };
  const prevPage = () => {
    setPage(page - 1);
  };

  const modifyVariable = (key: string) => (value: any): void => {
    const newVariables = { ...variables, [key]: value };
    setVariables(newVariables);
  };

  const validateFieldLength = (key: string) => (
    setCustomError: (error: CustomError) => void
  ) => (): boolean => {
    if ((variables[key as keyof typeof variables] as string).length === 0) {
      setCustomError(new CustomError(`${key} must not be empty`, [key]));
      return false;
    } else {
      setCustomError(new CustomError("", []));
      return true;
    }
  };

  return (
    <>
      <Layout style={styles.container}>
        <Layout style={styles.divider}>
          {page === 0 && (
            <EventDetailsForm
              variables={variables}
              modifyVariable={modifyVariable}
              nextPage={nextPage}
              validateFieldLength={validateFieldLength}
            />
          )}
          {page === 1 && (
            <EventDateForm
              variables={variables}
              modifyVariable={modifyVariable}
              nextPage={nextPage}
              prevPage={prevPage}
            />
          )}
          {page === 2 && (
            <EventLocationForm
              variables={variables}
              modifyVariable={modifyVariable}
              nextPage={nextPage}
              prevPage={prevPage}
              validateFieldLength={validateFieldLength}
            />
          )}
          {page === 3 && (
            <EventImagesForm
              variables={variables}
              modifyVariable={modifyVariable}
              nextPage={nextPage}
              prevPage={prevPage}
            />
          )}

          {page === 4 && (
            <EventCategoryForm
              variables={variables}
              modifyVariable={modifyVariable}
              nextPage={nextPage}
              prevPage={prevPage}
              validateFieldLength={validateFieldLength}
            />
          )}
          {page === 5 && (
            <SubmitForm
              variables={variables}
              modifyVariable={modifyVariable}
              prevPage={prevPage}
              role={role}
              id={props.event?.id}
            />
          )}
        </Layout>
      </Layout>
    </>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 20,

    flex: 1,
    flexDirection: "column",
    justifyContent: "center",
  },
  divider: {
    flexGrow: 1,
    justifyContent: "center",
  },
});

export default FormsHandler;
