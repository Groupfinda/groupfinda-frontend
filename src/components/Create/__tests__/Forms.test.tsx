import React from "react";
import {
  render,
  fireEvent,
  act,
  cleanup,
  RenderAPI,
} from "../../../../utils/test-utils";
import FormsHandler from "../Forms/FormsHandler";
import { wait } from "@apollo/react-testing";

jest.mock("@apollo/react-hooks", () => ({
  useApolloClient: () => ({
    readQuery: (_: any) => ({ me: { role: "USER" } }),
  }),
}));
test("renders event details UI correctly", () => {
  const tree = render(<FormsHandler />).toJSON();

  expect(tree).toMatchSnapshot();
});

const skipEventDetails = (component: RenderAPI) => {
  const titleInput = component.getByPlaceholder("Event title");
  const descriptionInput = component.getByPlaceholder(
    "Enter a description for your event"
  );
  fireEvent.changeText(titleInput, "something");
  fireEvent.changeText(descriptionInput, "something");

  pressNext(component);
};

const skipEventDate = (component: RenderAPI) => {
  pressNext(component);
};

const pressNext = (component: RenderAPI) => {
  const button = component.getByText("Next");
  fireEvent.press(button);
};

const pressPrev = (component: RenderAPI) => {
  const button = component.getByText("Prev");
  fireEvent.press(button);
};

test("event details renders text correctly", () => {
  const component = render(<FormsHandler />);
  expect(component.queryByText("Tell us more about your event.")).toBeTruthy();
  expect(component.queryByText("Title")).toBeTruthy();
  expect(
    component.queryByText("Write something to attract new participants")
  ).toBeTruthy();
  expect(component.queryByText("Description")).toBeTruthy();
  expect(
    component.queryByText("Let people know what your event is about")
  ).toBeTruthy();
  expect(component.queryByText("Next")).toBeTruthy();
});

test("event details renders title error correctly", () => {
  const component = render(<FormsHandler />);
  const titleInput = component.getByPlaceholder("Event title");
  const button = component.getByText("Next");
  fireEvent.press(button);
  expect(titleInput).toHaveProp("status", "danger");
  expect(component.queryByText("Title must not be empty")).toBeTruthy();

  fireEvent.changeText(titleInput, "something");
  expect(titleInput).toHaveProp("status", "basic");
  fireEvent.press(button);
  expect(component.queryByText("Title must not be empty")).not.toBeTruthy();
});

test("event details render description error correctly", () => {
  const component = render(<FormsHandler />);
  const titleInput = component.getByPlaceholder("Event title");
  const descriptionInput = component.getByPlaceholder(
    "Enter a description for your event"
  );
  const button = component.getByText("Next");

  fireEvent.changeText(titleInput, "something");
  fireEvent.press(button);
  expect(descriptionInput).toHaveProp("status", "danger");
  expect(component.queryByText("Description must not be empty")).toBeTruthy();
  fireEvent.changeText(descriptionInput, "something");
  expect(descriptionInput).toHaveProp("status", "basic");
});

test("cannot move on from event details if either title or description is empty", () => {
  const component = render(<FormsHandler />);
  const titleInput = component.getByPlaceholder("Event title");
  const descriptionInput = component.getByPlaceholder(
    "Enter a description for your event"
  );
  const button = component.getByText("Next");

  expect(component.queryByText("Tell us more about your event.")).toBeTruthy();

  fireEvent.press(button);
  expect(component.queryByText("Tell us more about your event.")).toBeTruthy();

  fireEvent.changeText(titleInput, "something");
  fireEvent.press(button);
  expect(component.queryByText("Tell us more about your event.")).toBeTruthy();

  fireEvent.changeText(titleInput, "");
  fireEvent.changeText(descriptionInput, "something");
  fireEvent.press(button);
  expect(component.queryByText("Tell us more about your event.")).toBeTruthy();
});

/** Cannot use this as date changes every day
test("renders event date ui correctly", () => {
  const component = render(<FormsHandler />);
  skipEventDetails(component);
  expect(component.toJSON()).toMatchSnapshot();
});
*/

test("event date renders text correctly", () => {
  const component = render(<FormsHandler />);
  skipEventDetails(component);

  expect(
    component.queryByText("When is this event taking place?")
  ).toBeTruthy();
  expect(component.queryByText("Date of Event")).toBeTruthy();
  expect(
    component.queryByText("This is the actual day of your event")
  ).toBeTruthy();
  expect(
    component.queryByText(
      "When is the last day for participants to be grouped up?"
    )
  ).toBeTruthy();
  expect(component.queryByText("Date of last register")).toBeTruthy();
  expect(
    component.queryByText("This is the last day for people to sign up")
  ).toBeTruthy();
  const dateString = new Date().toLocaleDateString();
  console.log(dateString);
  expect(component.queryByText(dateString)).toBeTruthy();
  expect(component.queryByText("Prev")).toBeTruthy();
  expect(component.queryByText("Next")).toBeTruthy();
});

test("renders event location ui correctly", () => {
  const component = render(<FormsHandler />);
  skipEventDetails(component);
  skipEventDate(component);
  expect(component.toJSON()).toMatchSnapshot();
});

test("event location renders text correctly", () => {
  const component = render(<FormsHandler />);
  skipEventDetails(component);
  skipEventDate(component);
  expect(
    component.queryByText("Where is this event taking place?")
  ).toBeTruthy();
  expect(component.queryByText("Address")).toBeTruthy();
  expect(
    component.queryByText("Let users know where your event is taking place")
  ).toBeTruthy();
  expect(component.queryByText("Postal Code")).toBeTruthy();

  expect(component.queryByText("Prev")).toBeTruthy();
  expect(component.queryByText("Next")).toBeTruthy();
});
