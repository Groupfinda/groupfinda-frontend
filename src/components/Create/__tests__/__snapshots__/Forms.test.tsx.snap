// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders event details UI correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    appearance="default"
    level="1"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
        },
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "padding": 20,
        },
      ]
    }
  >
    <View
      appearance="default"
      level="1"
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
          },
          Object {
            "flexGrow": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <View
          appearance="default"
          level="1"
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
              },
              undefined,
            ]
          }
        >
          <Text
            appearance="hint"
            category="h5"
            style={
              Array [
                Object {
                  "color": "#8F9BB3",
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "800",
                },
                Object {
                  "marginBottom": 20,
                },
              ]
            }
          >
            Tell us more about your event.
          </Text>
          <View
            style={
              Object {
                "marginBottom": 35,
                "marginTop": 10,
              }
            }
          >
            <Text
              appearance="default"
              category="p1"
              style={
                Array [
                  Object {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "400",
                  },
                  Array [
                    Object {
                      "color": "#8F9BB3",
                      "fontFamily": "System",
                      "fontSize": 12,
                      "fontWeight": "800",
                      "marginBottom": 4,
                    },
                    Object {
                      "textAlign": "left",
                    },
                  ],
                ]
              }
            >
              Title
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#F7F9FC",
                    "borderColor": "#E4E9F2",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "minHeight": 40,
                    "paddingHorizontal": 8,
                    "paddingVertical": 7,
                  },
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "width": "100%",
                  },
                ]
              }
            >
              <TextInput
                allowFontScaling={true}
                appearance="default"
                autoCapitalize="none"
                autoCorrect={false}
                blurOnSubmit={false}
                editable={true}
                onBlur={[Function]}
                onChange={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onSubmitEditing={[Function]}
                placeholder="Event title"
                placeholderTextColor="#8F9BB3"
                rejectResponderTermination={true}
                size="medium"
                status="basic"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "normal",
                      "marginHorizontal": 8,
                    },
                    Object {
                      "flexBasis": "auto",
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                    null,
                    undefined,
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "marginTop": 4,
                  },
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <Text
                appearance="default"
                category="p1"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "color": "#8F9BB3",
                        "fontFamily": "System",
                        "fontSize": 12,
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                    ],
                  ]
                }
              >
                Write something to attract new participants
              </Text>
            </View>
          </View>
          <View
            style={Object {}}
          >
            <Text
              appearance="default"
              category="p1"
              style={
                Array [
                  Object {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "400",
                  },
                  Array [
                    Object {
                      "color": "#8F9BB3",
                      "fontFamily": "System",
                      "fontSize": 12,
                      "fontWeight": "800",
                      "marginBottom": 4,
                    },
                    Object {
                      "textAlign": "left",
                    },
                  ],
                ]
              }
            >
              Description
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#F7F9FC",
                    "borderColor": "#E4E9F2",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "minHeight": 40,
                    "paddingHorizontal": 8,
                    "paddingVertical": 7,
                  },
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "width": "100%",
                  },
                ]
              }
            >
              <TextInput
                allowFontScaling={true}
                appearance="default"
                autoCapitalize="none"
                autoCorrect={false}
                editable={true}
                multiline={true}
                onBlur={[Function]}
                onChange={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onSubmitEditing={[Function]}
                placeholder="Enter a description for your event"
                placeholderTextColor="#8F9BB3"
                rejectResponderTermination={true}
                size="medium"
                status="basic"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "normal",
                      "marginHorizontal": 8,
                    },
                    Object {
                      "flexBasis": "auto",
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                    null,
                    Object {
                      "minHeight": 100,
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "marginTop": 4,
                  },
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <Text
                appearance="default"
                category="p1"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "color": "#8F9BB3",
                        "fontFamily": "System",
                        "fontSize": 12,
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                    ],
                  ]
                }
              >
                Let people know what your event is about
              </Text>
            </View>
          </View>
        </View>
        <View
          appearance="default"
          level="1"
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
              },
              Object {
                "flexDirection": "row",
                "marginTop": 20,
              },
            ]
          }
        >
          <View
            appearance="default"
            level="1"
            style={
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#008EED",
                "borderColor": "#008EED",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 40,
                "minWidth": 40,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              appearance="default"
              category="p1"
              style={
                Array [
                  Object {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "marginHorizontal": 10,
                  },
                ]
              }
            >
              Next
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders event location ui correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    appearance="default"
    level="1"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
        },
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "padding": 20,
        },
      ]
    }
  >
    <View
      appearance="default"
      level="1"
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
          },
          Object {
            "flexGrow": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        appearance="default"
        level="1"
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
            },
            Object {
              "flex": 1,
              "justifyContent": "space-around",
            },
          ]
        }
      >
        <View
          appearance="default"
          level="1"
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
              },
              Object {
                "marginVertical": 10,
              },
            ]
          }
        >
          <Text
            appearance="hint"
            category="h5"
            style={
              Array [
                Object {
                  "color": "#8F9BB3",
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "800",
                },
                Object {
                  "marginBottom": 20,
                },
              ]
            }
          >
            Where is this event taking place?
          </Text>
          <View
            style={
              Object {
                "marginBottom": 35,
                "marginTop": 10,
              }
            }
          >
            <Text
              appearance="default"
              category="p1"
              style={
                Array [
                  Object {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "400",
                  },
                  Array [
                    Object {
                      "color": "#8F9BB3",
                      "fontFamily": "System",
                      "fontSize": 12,
                      "fontWeight": "800",
                      "marginBottom": 4,
                    },
                    Object {
                      "textAlign": "left",
                    },
                  ],
                ]
              }
            >
              Address
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#F7F9FC",
                    "borderColor": "#E4E9F2",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "minHeight": 40,
                    "paddingHorizontal": 8,
                    "paddingVertical": 7,
                  },
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "width": "100%",
                  },
                ]
              }
            >
              <TextInput
                allowFontScaling={true}
                appearance="default"
                autoCapitalize="none"
                autoCorrect={false}
                blurOnSubmit={false}
                editable={true}
                onBlur={[Function]}
                onChange={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onSubmitEditing={[Function]}
                placeholder="Address of event"
                placeholderTextColor="#8F9BB3"
                rejectResponderTermination={true}
                size="medium"
                status="basic"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "normal",
                      "marginHorizontal": 8,
                    },
                    Object {
                      "flexBasis": "auto",
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                    null,
                    undefined,
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "marginTop": 4,
                  },
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <Text
                appearance="default"
                category="p1"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "color": "#8F9BB3",
                        "fontFamily": "System",
                        "fontSize": 12,
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                    ],
                  ]
                }
              >
                Let users know where your event is taking place
              </Text>
            </View>
          </View>
          <View
            style={Object {}}
          >
            <Text
              appearance="default"
              category="p1"
              style={
                Array [
                  Object {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "400",
                  },
                  Array [
                    Object {
                      "color": "#8F9BB3",
                      "fontFamily": "System",
                      "fontSize": 12,
                      "fontWeight": "800",
                      "marginBottom": 4,
                    },
                    Object {
                      "textAlign": "left",
                    },
                  ],
                ]
              }
            >
              Postal Code
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#F7F9FC",
                    "borderColor": "#E4E9F2",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "minHeight": 40,
                    "paddingHorizontal": 8,
                    "paddingVertical": 7,
                  },
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "width": "100%",
                  },
                ]
              }
            >
              <TextInput
                allowFontScaling={true}
                appearance="default"
                autoCapitalize="none"
                autoCorrect={false}
                editable={true}
                onBlur={[Function]}
                onChange={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onSubmitEditing={[Function]}
                placeholder="Enter postal code"
                placeholderTextColor="#8F9BB3"
                rejectResponderTermination={true}
                size="medium"
                status="basic"
                style={
                  Array [
                    Object {
                      "color": "#222B45",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "normal",
                      "marginHorizontal": 8,
                    },
                    Object {
                      "flexBasis": "auto",
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                    null,
                    undefined,
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "marginTop": 4,
                  },
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                ]
              }
            />
          </View>
        </View>
        <View
          appearance="default"
          level="1"
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
              },
              Object {
                "flexDirection": "row",
                "marginTop": 20,
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#008EED",
                "borderColor": "#008EED",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 40,
                "minWidth": 40,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              appearance="default"
              category="p1"
              style={
                Array [
                  Object {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "marginHorizontal": 10,
                  },
                ]
              }
            >
              Prev
            </Text>
          </View>
          <View
            appearance="default"
            level="1"
            style={
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#008EED",
                "borderColor": "#008EED",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 40,
                "minWidth": 40,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              appearance="default"
              category="p1"
              style={
                Array [
                  Object {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "marginHorizontal": 10,
                  },
                ]
              }
            >
              Next
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
